app/wama.web/Controllers/CheckInController.cs

```diff
@@ -60,6 +71,30 @@ public IActionResult WaiverNotSigned()
             return View();
         }
 
+        [HttpPost]
+        public async Task<IActionResult> Waiver(string signerName, string memberId)
+        {
+            var userAccount = await _UserAccountService.GetUserAccountAsync(memberId);
```

Comment:
Program execution shouldn't come to the Waiver page if the userAccount is returned null for that given memberid.